### PR TITLE
create a new contact UI

### LINKED ISSUE #5: Create a new contact UI

### PR DESCRIPTION
**Pull request by abhishekt98**
Fixes #5 

- New contact UI added
- Delete contact UI added

### CODE REVIEW

File: app/javascript/src/components/contacts/ContactTable.jsx [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+  };
+
+  const iconColumn = () => {
+    return (
+      <Dropdown icon={MenuVertical} buttonStyle="text" position="bottom-end">
+        <li>Edit</li>
+        <li onClick={() => setShowDeleteAlert(true)}>Delete</li>
+      </Dropdown>
+    );
+  };
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/ContactTable.jsx:31]: IMO, there's no need to do this. If you wish to, please use better names.

File: app/javascript/src/components/contacts/Menubar.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+
+import { Search, Settings, Plus } from "@bigbinary/neeto-icons";
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/Menubar.jsx:3]: Please use aliases.

File: app/javascript/src/components/contacts/Menubar.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import { Typography } from "neetoui/v2";
+import { MenuBar } from "neetoui/v2/layouts";
+
+const ContactMenubar = ({ title }) => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/Menubar.jsx:7]: Please name this as `MenuBar` also please name filenames and folders properly.

File: app/javascript/src/components/contacts/index.jsx [hunk 4]
```diff
@@ -1,22 +1,41 @@
...
+import Header from "components/common/header";
+import ContactTable from "components/contacts/ContactTable";
+import ContactMenuBar from "components/contacts/Menubar";
+import AddContact from "components/contacts/pane/AddContact";
+import Container from "components/Container";
 
 const Index = () => {
   const contacts = data.contacts;
+  const [showPane, setShowPane] = useState(false);
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/index.jsx:15]:
```jsx
const [isNewContactPaneOpen, setIsNewContactPaneOpen] = useState(false);
const [isDeleteAlertOpen, setIsDeleteAlertOpen] = useState(false);
```

File: app/javascript/src/components/contacts/pane/AddContact.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <Typography style="h2" weight="semibold">
+          Create a New Contact
+        </Typography>
+      </Pane.Header>
+      <ContactForm onClose={onClose} contact={formInitialValues.contactsForm} />
+    </Pane>
+  );
+};
+
+export default Addcontact;
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/AddContact.jsx:23]: We don't need a folder called `pane`.  Also, please provide proper folder names.

File: app/javascript/src/components/contacts/pane/AddContact.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const onClose = () => setShowPane(false);
+
+  return (
+    <Pane isOpen={showPane} onClose={onClose}>
+      <Pane.Header>
+        <Typography style="h2" weight="semibold">
+          Create a New Contact
+        </Typography>
+      </Pane.Header>
+      <ContactForm onClose={onClose} contact={formInitialValues.contactsForm} />
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/AddContact.jsx:18]: Please don't suffix component or filenames with `Modal, Form, Pane or Page`.

File: app/javascript/src/components/contacts/pane/ContactForm.jsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    >
+      {({ isSubmitting, handleSubmit }) => (
+        <Form>
+          <Pane.Body>
+            <div className="w-full space-y-6">
+              <div className="grid grid-cols-2 gap-4">
+                <Input
+                  label="First Name"
+                  name="firstName"
+                  required
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/ContactForm.jsx:33]: Do we need required here?

File: app/javascript/src/components/contacts/pane/ContactForm.jsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                  size="large"
+                />
+                <Input
+                  label="Last Name"
+                  name="lastName"
+                  required
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/ContactForm.jsx:39]: Do we need required here?

File: app/javascript/src/components/contacts/pane/ContactForm.jsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+                  size="large"
+                />
+              </div>
+
+              <Input
+                label="Email"
+                name="email"
+                className="flex-grow-0 w-full"
+                required
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/ContactForm.jsx:48]: Do we need required here?

File: app/javascript/src/components/contacts/pane/ContactForm.jsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+                size="large"
+              />
+              <Select
+                name="role"
+                label="Role"
+                placeholder="Select Role"
+                options={[
+                  { label: "option1", value: "value1" },
+                  { label: "option2", value: "value2" },
+                ]}
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/ContactForm.jsx:58]: Please move to a constants file in the scope of this directory and use proper labels.

File: app/javascript/src/components/contacts/pane/ContactForm.jsx [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+                required
```
Comment (karthiknmenon) [inline app/javascript/src/components/contacts/pane/ContactForm.jsx:59]: Do we need required here?

File: app/javascript/src/components/notes/Menubar.jsx [hunk 12]
```diff
@@ -4,7 +4,7 @@ import { Search, Settings, Plus } from "@bigbinary/neeto-icons";
 import { Typography } from "neetoui/v2";
 import { MenuBar } from "neetoui/v2/layouts";
 
-const Index = ({ title }) => {
+const NoteMenubar = ({ title }) => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/Menubar.jsx:7]: `MenuBar`

File: app/javascript/src/components/notes/index.jsx [hunk 13]
```diff
@@ -1,28 +1,28 @@
 import React, { useState } from "react";
 
+import data from "data";
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/index.jsx:3]: Why do we need this here?

File: app/javascript/src/components/notes/index.jsx [hunk 14]
```diff
@@ -1,28 +1,28 @@
...
-import Header from "components/header";
-import MenuBar from "components/menubar";
-import DeleteAlert from "components/notes/DeleteAlert";
+import NoteMenuBar from "components/notes/Menubar";
 import Note from "components/notes/note";
 import Addnote from "components/notes/pane/AddNote";
 
-import data from "../../data";
-
 const Index = () => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/index.jsx:12]: Please don't use `index` as a component name.

File: app/javascript/src/components/notes/index.jsx [hunk 15]
```diff
@@ -1,28 +1,28 @@
...
   const notes = data.notes;
   const [showDeleteAlert, setShowDeleteAlert] = useState(false);
   const [showPane, setShowPane] = useState(false);
+
   return (
     <Container>
-      <MenuBar title={"Notes"} />
+      <NoteMenuBar title={"Notes"} />
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/index.jsx:19]: We don't need to use `{}` with strings.

File: app/javascript/src/components/notes/index.jsx [hunk 16]
```diff
@@ -40,6 +40,7 @@ const Index = () => {
       </div>
       <Addnote showPane={showPane} setShowPane={setShowPane} />
       <DeleteAlert
+        entity={"Note"}
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/index.jsx:43]: Same here.

File: app/javascript/src/components/notes/pane/NoteForm.jsx [hunk 17]
```diff
@@ -25,16 +26,10 @@ const NoteForm = ({ onClose, note }) => {
         <Form>
           <Pane.Body>
             <div className="w-full space-y-6">
-              <Input
-                label="Title"
-                name="title"
-                className="flex-grow-0 w-full"
-                required
-              />
+              <Input label="Title" name="title" required />
```
Comment (karthiknmenon) [inline app/javascript/src/components/notes/pane/NoteForm.jsx:29]: Do we need required here?

File: app/javascript/src/constants/formInitialValues.js [hunk 18]
```diff
@@ -5,4 +5,10 @@ export default {
     contact: null,
     tags: [],
   },
+  contactsForm: {
+    firstName: "",
+    lastName: "",
+    email: "",
+    role: "",
+  },
```
Comment (karthiknmenon) [inline app/javascript/src/constants/formInitialValues.js:13]: Please move this to a constants file in the scope of `Contacts` directory.

File: app/javascript/src/constants/formValidationSchemas.js [hunk 19]
```diff
@@ -7,4 +7,12 @@ export default {
...
     contact: Yup.object().required("Assigned Contact is required").nullable(),
   }),
+  contactsForm: Yup.object().shape({
+    firstName: Yup.string().required("First Name is required"),
+    lastName: Yup.string().required("Last Name is required"),
+    email: Yup.string()
+      .email("Invalid email address")
+      .required("Email is Required"),
+    role: Yup.object().required("Role is required"),
+  }),
```
Comment (karthiknmenon) [inline app/javascript/src/constants/formValidationSchemas.js:17]: Please move this to a constants file in the scope of this directory.

**Review by karthiknmenon — COMMENTED** [review]
@abhishekt98 Please check the suggestions.

File: app/javascript/src/components/notes/pane/NoteForm.jsx (hunk 17 above)
Comment (abhishekt98) [reply app/javascript/src/components/notes/pane/NoteForm.jsx:29]: All the fields in the contact form is required field in the given figma design.

File: app/javascript/src/components/notes/index.jsx (hunk 13 above)
Comment (abhishekt98) [reply app/javascript/src/components/notes/index.jsx:3]: I stored the dummy notes data in  `data` folder . Now I moved the sample data into respective `constants.js` file.

File: app/javascript/src/components/notes/Menubar.jsx (hunk 12 above)
Comment (abhishekt98) [reply app/javascript/src/components/notes/Menubar.jsx:7]: I can't use `MenuBar` as component name since the `neetoui` component name is also `MenuBar`

**Pull request merged**